src/doc/unstable-book/src/language-features/macro-metavar-expr-concat.md

```diff
@@ -0,0 +1,152 @@
+# `macro_metavar_expr_concat`
+
+The tracking issue for this feature is: [#124225]
+
+------------------------
+
+In stable Rust, there is no way to create new identifiers by joining identifiers to literals or other identifiers without using procedural macros such as [`paste`].
+ `#![feature(macro_metavar_expr_concat)]` introduces a way to do this, using the concat metavariable expression.
+
+> This feature is not to be confused with [`macro_metavar_expr`] or [`concat_idents`].
+
+> This is an experimental feature; it and its syntax will require a RFC before stabilization.
+
+
+### Overview
+
+At this time, [declarative macros] cannot create new identifiers.
+A common use case is the need to create names for structs or functions. The following cannot be done on stable Rust[^1]:
+
+```rust,compile_fail
+macro_rules! create_some_structs {
+    ($name:ident) => {
+        // Invalid syntax
+        pub struct First$name;
+        // Also invalid syntax
+        pub struct Second($name);
+        // Macros are not allowed in this position
+        // (This restriction is what makes `concat_idents!` useless)
+        pub struct concat_idents!(Third, $name);
+    }
+}
+# create_some_structs!(Thing);
+```
```

Comment:
This bit can be eliminated, `macro_metavar_expr_concat` shouldn't be discussed in terms of `concat_idents`'s limitations.